File: rust/49_group_anagrams.rs
impl Solution {
    pub fn group_anagrams(strs: Vec<String>) -> Vec<Vec<String>> {
        use std::collections::HashMap;
        let mut key_map = HashMap::<Vec<u8>, Vec<String>>::with_capacity(strs.len());

        for s in strs {
            let mut key = s.clone().into_bytes();
            key.sort();
            key_map.entry(key).or_default().push(s);
        }

        key_map.into_values().collect::<Vec<Vec<String>>>()
    }
}

File: cpp/49_group_anagrams.cpp
#include <string>
#include <unordered_map>
#include <vector>

using namespace std;

class Solution {
 public:
  vector<vector<string>> groupAnagrams(vector<string> &strs) {
    unordered_map<string, vector<string>> map;
    for (const auto &s : strs) {
      string key = s;
      sort(key.begin(), key.end());
      map[key].push_back(s);
    }
    vector<vector<string>> result;
    for (auto &[key, words] : map) {
      result.push_back(std::move(words));
    }
    return result;
  }
};
